Test/Unit/Context.php

```diff
@@ -0,0 +1,226 @@
+<?php
+
+/**
+ * Hoa
+ *
+ *
+ * @license
+ *
+ * New BSD License
+ *
+ * Copyright © 2007-2015, Hoa community. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *     * Redistributions of source code must retain the above copyright
+ *       notice, this list of conditions and the following disclaimer.
+ *     * Redistributions in binary form must reproduce the above copyright
+ *       notice, this list of conditions and the following disclaimer in the
+ *       documentation and/or other materials provided with the distribution.
+ *     * Neither the name of the Hoa nor the names of its contributors may be
+ *       used to endorse or promote products derived from this software without
+ *       specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDERS AND CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+namespace Hoa\Math\Test\Unit;
+
+use Hoa\Core;
+use Hoa\Math as LUT;
+use Hoa\Math\Context as CUT;
+use Hoa\Test;
+
+/**
+ * Class \Hoa\Math\Test\Unit\Context.
+ *
+ * Test suite of the Hoa\Math\Context class.
+ *
+ * @copyright  Copyright © 2007-2015 Hoa community
+ * @license    New BSD License
+ */
+class Context extends Test\Unit\Suite
+{
+    public function case_context_has_no_predefined_variable()
+    {
+        $this
+            ->given($context = new CUT())
+            ->then
+                ->array(iterator_to_array($context->getVariables()))->isEmpty();
+    }
+
+    public function case_context_exception_when_getting_unknown_variable()
+    {
+        $this
+            ->given(
+                $name    = 'foo',
+                $context = new CUT()
+            )
+            ->then
+                ->exception(function () use ($context, $name) {
+                    $context->getVariable($name);
+                })
+                    ->isInstanceOf('Hoa\Math\Exception\UnknownVariable')
+                    ->hasMessage('Variable ' . $name .' does not exist');
```

Comment:
Not sure if this is useful to test message. We used to no test them. I guess you can remove `hasMessage` assertion*s*.